secure/storage/src/vault.rs

```diff
@@ -223,18 +231,32 @@ impl KVStorage for VaultStorage {
         let resp = self.client().read_secret(&secret, key)?;
         let last_update = DateTime::parse_from_rfc3339(&resp.creation_time)?.timestamp() as u64;
         let value: T = serde_json::from_value(resp.value)?;
+        self.secret_versions
+            .write()
+            .unwrap()
+            .insert(key.to_string(), resp.version);
         Ok(GetResponse { last_update, value })
     }
 
     fn set<T: Serialize>(&mut self, key: &str, value: T) -> Result<(), Error> {
         let secret = self.secret_name(key);
-        self.client()
-            .write_secret(&secret, key, &serde_json::to_value(&value)?)?;
+        let version = match self.secret_versions.read().unwrap().get(&key.to_string()) {
```

Comment:
&key.to_string :O
that seems a little weird...
yeah, you can just use the key:
https://github.com/libra/libra/blob/master/secure/storage/src/in_memory.rs#L46